.adops/package-update.yml

```diff
@@ -20,16 +20,22 @@ variables:
     value: /home/vsts/.yarn/berry/cache
   - name: buildSourceBranch
     value: $[replace(variables['Build.SourceBranch'], 'refs/heads/', '')]
+  - name: temporaryPushBranch
+    value: 'chore/update-mosaic-libs-pipeline-branch-$(Build.BuildNumber)'
+  - name: permanentPushBranch
+    value: 'main'
 
 steps:
   - checkout: self
-    clean: 'true'
     persistCredentials: 'true'
 
   - script: |
+      git fetch
       git config user.email "no-email@axinom.com"
       git config user.name "Mosaic CI"
-    displayName: Set git identity
+      echo "Check out to target branch: $(permanentPushBranch)"
+      git checkout $(permanentPushBranch) && git pull origin $(permanentPushBranch)
```

Comment:
pull and checkout should be switched?

Reply:
@EmperorRXF `git fetch` fetches the remote branches and tags and then we check into the main branch; `git pull origin main` is redundant since the pipeline anyway clones the most recent version of the repository.

Reply:
right yes - then we can remove the `&& git pull`